src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -1357,13 +1357,7 @@ internal static bool StrEqual(char[]? chars, int strPos1, int strLen1, string st
 
             Debug.Assert(chars != null);
 
-            int i = 0;
-            while (i < strLen1 && chars[strPos1 + i] == str2[i])
-            {
-                i++;
-            }
-
-            return i == strLen1;
+            return chars.AsSpan(strPos1, strLen1).SequenceEqual(str2.AsSpan());
```

Comment:
Rather than changing this method, can we instead delete this method and just use AsSpan.SequenceEqual at the call sites? In many cases it looks like it could then instead be simplified to be AsSpan.StartsWith, e.g. instead of:
```C#
if (XmlConvert.StrEqual(chars, pos, 6, "CDATA["))
```
it'd be:
```C#
if (chars.AsSpan(pos).StartsWith("CDATA["))
```
which is then also more likely to pick up the JIT optimizations around unrolling the comparison against constant strings.

Reply:
Sure, just changes will be bigger - current method is used in 39 places.
I will adapt PR as suggested - using .StartsWith